link/syscalls.go

```diff
@@ -26,3 +31,55 @@ var haveProgAttach = internal.FeatureTest("BPF_PROG_ATTACH", "4.10", func() (boo
 	prog.Close()
 	return true, nil
 })
+
+type bpfLinkCreateAttr struct {
+	progFd     uint32
+	targetFd   uint32
+	attachType ebpf.AttachType
+	flags      uint32
+}
+
+func bpfLinkCreate(attr *bpfLinkCreateAttr) (*internal.FD, error) {
+	ptr, err := internal.BPF(internal.BPF_LINK_CREATE, unsafe.Pointer(attr), unsafe.Sizeof(*attr))
+	if err == nil {
```

Comment:
seems a bit odd to `if good_case {return good} return bad` instead of `if  bad {bad} good;` but just a nit take it or leave it.

Reply:
Good point, fixed